test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/DataPartitionEnabledTests.cs

```diff
@@ -57,18 +57,35 @@ public async Task WhenRetrievingPartitions_TheServerShouldReturnAllPartitions()
     public async Task GivenDatasetWithNewPartitionName_WhenStoring_TheServerShouldReturnWithNewPartition()
     {
         var newPartition = new Partition(Partition.DefaultKey, TestUidGenerator.Generate());
+        await ValidateParititonCreation(newPartition);
+    }
 
+    [Fact]
+    [Trait("Category", "bvt-dp")]
+    public async Task GivenDatasetWithNewPartitionName_WhenStoringInparallel_TheServerShouldReturnWithNewPartition()
+    {
+        var newPartition = new Partition(Partition.DefaultKey, TestUidGenerator.Generate());
+
+        await Task.WhenAll(
+            ValidateParititonCreation(newPartition),
+            ValidateParititonCreation(newPartition),
+            ValidateParititonCreation(newPartition));
+    }
+
+    private async Task<DicomWebResponse<DicomDataset>> ValidateParititonCreation(Partition newPartition)
+    {
         string studyInstanceUID = TestUidGenerator.Generate();
 
         DicomFile dicomFile = Samples.CreateRandomDicomFile(studyInstanceUID);
 
-        using DicomWebResponse<DicomDataset> response = await _instancesManager.StoreAsync(new[] { dicomFile }, partition: newPartition);
+        DicomWebResponse<DicomDataset> response = await _instancesManager.StoreAsync(new[] { dicomFile }, partition: newPartition);
 
         Assert.True(response.IsSuccessStatusCode);
 
         ValidationHelpers.ValidateReferencedSopSequence(
             await response.GetValueAsync(),
             ConvertToReferencedSopSequenceEntry(dicomFile.Dataset, newPartition.Name));
+        return response;
```

Comment:
Why do we want to return response now?

Reply:
There is another test that is also using the same code. check line 60.

Reply:
I get that but that test is not using the response value right?